source/_posts/2016-11-7-samg55-oled1-xplained-pro.markdown

```diff
@@ -0,0 +1,158 @@
+will do that for you. Open ASF Wizard, and find there
+ ```SSD1306 OLED controller (component)```
+select it, and apply changes.
+
+<a class="fancybox" rel="group" href="/assets/images/Capture.PNG"><img
+src="/assets/images/Capture.PNG" alt="" /></a>
+
+ Now your Solution Explorer got few more files.  You may add simple chunk of
+code in the main function:
+
+```
+ssd1306_init();
+ssd1306_display_on();
+ssd1306_clear();
+ssd1306_set_column_address(40);
+ssd1306_set_page_address(2);
+ssd1306_write_text("Hello World");
+```
+
+But this will not work yet, you sill need to do some configuration.  Both files
+to change you can find in `config/` folder first one is `conf_board.h` In there
+you have to add these lines:
+
+```
+#define BOARD_FLEXCOM_SPI FLEXCOM5
+#define CONF_BOARD_OLED_UG_2832HSWEG04
+#define CONF_BOARD_SPI
+#define CONF_BOARD_SPI_NPCS1
+```
+
+Second one is `conf_ssd1306.h` In which you have to change:
+
+```
+# define SSD1306_SPI SPI5
+# define SSD1306_DC_PIN UG_2832HSWEG04_DATA_CMD_GPIO
+# define SSD1306_RES_PIN UG_2832HSWEG04_RESET_GPIO
+# define SSD1306_CS_PIN UG_2832HSWEG04_SS
+```
+
+Note, that these values are there twice, one time in `if`, that check whether
+your board is XMEGA_C3_XPLAINED or XMEGA_E5_XPLAINED, if it is, then change
+these values. For every other board, values can be found at the end of the
+file.  In the same place you will find comment explaining their meaning.
+comment.  After these changes, all you have to do is connect the board, using
+microUSB and connecting it to EDBG USB port, wait for Atmel Studio to find
+board, select tool `EDBG`, interface `SWD` and program the chip. After short
+amount of time, you will see "Hello World" on display, and blinking led.  To
+make it show whether function is active or inactive, change last while loop in
+main.c to something like this
+
+```
+while (1) {
+    if (g_b_led0_active) {
+        ioport_toggle_pin_level(LED0_GPIO);
+        ssd1306_clear();
+        ssd1306_set_column_address(40);
+        ssd1306_set_page_address(2);
+        ssd1306_write_text("Function is active!");
+        printf("1 ");
+    }else {
+        ssd1306_clear();
+        ssd1306_set_column_address(40);
+        ssd1306_set_page_address(2);
+        ssd1306_write_text("Function is inactive!");
+    }
+    mdelay(500);
+}
+```
+
+You might have noticed that `printf("1 ");`, and was wondering where you can
+find it's output? Serial console can be opened using Data Visualizer wich is in
+tools menu (if you have it installed), on the left side of it is
+`configuration` option, that will open panel, with several options to chose
+terminal can be opened selecting `External Connection` and `Serial Port`.
+Before connecting, remember to change baudrate to `115200`.  Now you are all
+set up, and ready to code.
+
+Sources
+-------
+
+SAMG55 Xplained Pro documentation
+ 
+[Data Gateway Interface](http://www.atmel.com/Images/Atmel-32223-Data-Gateway-Interface_UserGuide.pdf)
+
```

Comment:
@jakubruczynski no need of empty line here - looks ugly after rendering. Please also add bullets for those links.